WcaOnRails/app/views/competitions/_nearby_competitions.html.erb

```diff
@@ -11,6 +11,9 @@
   <div class="col-sm-9">
     <% if @competition.has_date? && @competition.has_location? %>
       <% nearby_competitions = @competition.nearby_competitions(Competition::NEARBY_DAYS_WARNING, Competition::NEARBY_DISTANCE_KM_WARNING)[0, 10] %>
+      <% if !current_user.can_view_hidden_competitions? %>
```

Comment:
This might be outside of the scope for this review, but to me this whole blob (lines 12-30) of complex logic should be in the controller.

`Competition. nearby_competitions` could take `current_user` as an optional parameter without violating any architectural principles I care about.